File: Tyche.Android/pool/src/main/java/com/felipearpa/pool/application/GetPoolUseCase.kt
package com.felipearpa.pool.application

import com.felipearpa.pool.domain.PoolRepository
import javax.inject.Inject

class GetPoolUseCase @Inject constructor(private val poolRepository: PoolRepository) {

    suspend fun execute(poolId: String) =
        poolRepository.getPool(poolId = poolId)
}

File: iOS/DataPool/Sources/DataPool/Application/GetPoolUseCase.swift
import Core

public class GetPoolUseCase {
    private let poolRepository: PoolRepository

    public init(poolRepository: PoolRepository) {
        self.poolRepository = poolRepository
    }

    public func execute(poolId: String) async -> Result<Pool, Error> {
        return await poolRepository.getPool(id: poolId)
    }
}
